conjure-go-client/httpclient/body_handler.go

```diff
@@ -82,8 +82,9 @@ func (b *bodyMiddleware) setRequestBody(req *http.Request) error {
 	if buf.Len() != 0 {
 		req.Body = ioutil.NopCloser(buf)
 		req.ContentLength = int64(buf.Len())
-		roCopy := ioutil.NopCloser(bytes.NewReader(buf.Bytes()))
-		req.GetBody = func() (io.ReadCloser, error) { return roCopy, nil }
+		req.GetBody = func() (io.ReadCloser, error) {
+			return ioutil.NopCloser(bytes.NewReader(buf.Bytes())), nil
```

Comment:
Trying to work through how this interacts with the optional buffer pool. Is it a problem for this to hold a pointer to the buffer, which might be returned to the pool in the defer above?

Reply:
I actually didn't see that previously -- thanks for pointing it out. I'm not sure how the bufferPool is at play here, let me do some digging.

Reply:
@bmoylan I took a look at our use of bufferpool and it looks like there's some adjacent issues there. I'm going to open a new PR to address those first!

Reply:
Namely, we aren't using the bufferpool currently :') womp

Reply:
lol wonderful... thanks for opening #93 let's try to get that in before continuing on this one

Reply:
Should we add a test with a buffer pool to make sure this works as intended now that we've fixed them?

Reply:
I had typed out this comment yesterday, but hadn't submitted it yet:

cool, so I merged the change from #93. Given that we only return the buffer pointer to the pool after finishing the `RoundTrip`, I don't think there's any issues with storing the pointer in the `GetBody` function. We were also previously storing the pointer in the Body as well: ` req.Body = ioutil.NopCloser(buf)`.